File: CleanCodeJava/app/src/main/java/io/chuumong/cleancodejava/di/AppModule.java
package io.chuumong.cleancodejava.di;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;
import io.chuumong.cleancodejava.App;
import io.chuumong.cleancodejava.data.api.CityService;
import retrofit2.Retrofit;
import retrofit2.http.POST;

/**
 * Created by JongHunLee on 2017-02-22.
 */
@Module
public class AppModule {

    private final App app;

    public AppModule(App app) {
        this.app = app;
    }

    @Provides
    @Singleton
    App provideApp() {
        return app;
    }

    @Provides
    @Singleton
    CityService provideCityService(Retrofit retrofit) {
        return retrofit.create(CityService.class);
    }
}

File: CleanCodeKotlin/app/src/main/java/io/chuumong/cleancodekotlin/di/AppModule.kt
package io.chuumong.cleancodekotlin.di

import dagger.Module
import dagger.Provides
import io.chuumong.cleancodekotlin.App
import io.chuumong.cleancodekotlin.data.api.CityService
import retrofit2.Retrofit
import javax.inject.Singleton

/**
 * Created by JongHunLee on 2017-02-23.
 */

@Module
class AppModule(val app: App) {

    @Provides
    @Singleton
    fun provideApp() = app

    @Provides
    @Singleton
    fun provideCityService(retrofit: Retrofit) = retrofit.create(CityService::class.java)
}
